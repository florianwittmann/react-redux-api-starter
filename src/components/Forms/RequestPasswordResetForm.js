import React from "react";
import { reduxForm } from "redux-form";
import { connect } from "react-redux";
import InputField from "./components/InputField";
import Alert from "./components/Alert";
import SubmitButton from "./components/SubmitButton";
import {
  isRequired,
  isEmail,
  validate as checkValidation
} from "./components/Validations";
import { requestPasswordReset } from "../../data/user/actions";
import { withRouter } from "react-router-dom";

export const RequestPasswordResetForm = props => {
  const onSubmit = ({ email }) => {
    props.requestPasswordReset({ email }, () => {
      props.history.push("/reset/success");
    });
  };

  return (
    <form onSubmit={props.handleSubmit(onSubmit)}>
      <InputField type="email" name="email" title="Email" icon="envelope" />
      <Alert errorMessage={props.errorMessage} />
      <SubmitButton inProgress={props.inProgress}>Continue</SubmitButton>
    </form>
  );
};

function validate(values) {
  return checkValidation(values, {
    email: [isRequired, isEmail]
  });
}

function mapStateToProps(state) {
  return { errorMessage: state.user.error, inProgress: state.user.inProgress };
}

const reduxFormRequestPasswordReset = reduxForm({
  validate,
  form: "RequestPasswordResetForm"
})(RequestPasswordResetForm);
export default withRouter(
  connect(mapStateToProps, { requestPasswordReset })(
    reduxFormRequestPasswordReset
  )
);
