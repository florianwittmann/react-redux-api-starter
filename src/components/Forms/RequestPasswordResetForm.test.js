import React from "react";
import { shallow } from "enzyme";
import { RequestPasswordResetForm } from "./RequestPasswordResetForm";

test("should render the RequestPasswordResetForm", () => {
  const wrapper = shallow(<RequestPasswordResetForm handleSubmit={() => {}} />);
  expect(wrapper).toMatchSnapshot();
});
