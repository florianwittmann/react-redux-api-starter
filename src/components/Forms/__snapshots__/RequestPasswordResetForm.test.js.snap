// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the RequestPasswordResetForm 1`] = `
<form>
  <InputField
    icon="envelope"
    name="email"
    title="Email"
    type="email"
  />
  <Component />
  <Component>
    Continue
  </Component>
</form>
`;
